README.md

```diff
@@ -27,115 +31,164 @@ repositories' pull requests by using the GitHub Pull Request Builder plugin. It
-GitHub automatically, the slaves must have be able to identify the hosts. Add
-them to the "known_hosts" file by executing a simple ssh command from the slaves
-to the upload server and to github.com. Alternatively, you could disable
-"StrictHostKeyChecking", but that may be a security hazard.
+After the SSH proxy is configured, simply use the following to connect to the system:
+`ssh <myuser>@ophost-builds`
 
-### Execute Ansible playbooks
+### Setup SSH credentials used to access upload server and GitHub from Jenkins slaves
 
-There are Ansible playbooks that set up the Jenkins master node and two Jenkins
-slave nodes, one for Host OS builds and one for validating the code from GitHub
-pull requests. There could be multiple slave nodes set up for parallel builds.
-The different roles (master, builds slave and validation slave) could even be
-set up in the same system, as desired.
+To automate SSH authentication between 2 hosts, it is necessary to:
+ - create a SSH key pair if you still do not have one
+ - add the target host to ~/.ssh/known_hosts in the source host. It is also
+   possible to do it by executing a simple SSH command from the source host to
+   the target host. Alternatively, you could disable "StrictHostKeyChecking"
+   for all SSH connections in the source host, but that may be a security hazard.
+ - add the SSH private key to ~/.ssh/ in the source host
+ - add the SSH public key to ~/.ssh/authorized_keys file in the target host
 
-Read the [Ansible instructions](ansible/README.md) for details on how to execute
+To be able to upload the results of builds to a server and push commits to
+GitHub automatically, you must automate SSH authentication from Jenkins slaves
+to those two servers.
+
+### Automatically setup Jenkins master and slave(s) using Ansible playbooks
+
+There are two Ansible playbooks: one that sets up a Jenkins master node and
+another one that sets up a Jenkins slave node. Read the
+[Ansible instructions](ansible/README.md) for details on how to execute
 the playbooks.
 
-Note: The "master" playbook may fail due to Jenkins temporary unresponsiveness.
-If you see HTTP request errors, try running it again.
+If you wish to have a single system hosting the entire Jenkins instance, both
+playbooks can be executed in the same system. Execute first the Jenkins master
+playbook, stop Jenkins service (systemctl stop jenkins) and then execute the
+Jenkins slave playbook.
+
+Note: The Jenkins playbooks may fail due to network errors. If you see HTTP
+request errors, try executing them again.
+
 
-### Manual Jenkins steps
+### Manually setup Jenkins via web UI
+
+If you do not have a direct network connection to the system(s), you may
+create a SSH tunnel in your workstation to be able to access the Jenkins
+web UI, which is accessible via HTTPS. Example:
+
+`sudo ssh -o "ProxyCommand ssh <user>@<proxy_server> -W %h:%p" -NL 443:localhost:443 root@<jenkins_master>`
```

Comment:
proxy server is the server which will be the intermediate (proxy) between you and the Jenkins server. I'll make it explicit in the doc.